cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSICartoDBSink.java

```diff
@@ -398,4 +444,138 @@ protected String buildTableName(String servicePath, String entity, String attrib
+        backend.insert(tableName, withs, fields, rows);
+    } // persistRawAggregation
+    
+    private void persistDistanceEvent(NGSIEvent event) throws Exception {
+        // Get some values
+        String servicePath = event.getServicePath();
+        String entityId = event.getContextElement().getId();
+        String entityType = event.getContextElement().getType();
+        
+        // Iterate on all this context element attributes, if there are attributes
+        ArrayList<ContextAttribute> contextAttributes = event.getContextElement().getAttributes();
+
+        if (contextAttributes == null || contextAttributes.isEmpty()) {
+            return;
+        } // if
+
+        for (ContextAttribute contextAttribute : contextAttributes) {
+            long recvTimeMs = event.getRecvTimeTs();
+            String attrType = contextAttribute.getType();
+            String attrValue = contextAttribute.getContextValue(false);
+            String attrMetadata = contextAttribute.getContextMetadata();
+            String location = NGSIUtils.getLocation(attrValue, attrType, attrMetadata, flipCoordinates);
+            String tableName = schema
+                    + "." + buildTableName(event.getServicePath(), event.getEntity(), event.getAttribute())
+                    + "_distance";
+
+            if (location.startsWith("ST_SetSRID(ST_MakePoint(")) {
+                // Try creating the table... the cost of checking if it exists and creating it is higher than directly
+                // attempting to create it. If existing, nothing will be re-created and the new values will be inserted
+                
+                try {
+                    String typedFields = "(recvTimeMs bigint, fiwareServicePath text, entityId text, entityType text, "
+                            + "stageDistance float, stageTime float, stageSpeed float, sumDistance float, "
+                            + "sumTime float, sumSpeed float, sum2Distance float, sum2Time float, sum2Speed float, "
+                            + "maxDistance float, minDistance float, maxTime float, minTime float, maxSpeed float, "
+                            + "minSpeed float, numSamples bigint)";
+                    backend.createTable(schema, tableName, typedFields);
+                    
+                    // Once created, insert the first row
+                    String withs = "";
+                    String fields = "(recvTimeMs, fiwareServicePath, entityId, entityType, the_geom, stageDistance,"
+                            + "stageTime, stageSpeed, sumDistance, sumTime, sumSpeed, sum2Distance, sum2Time,"
+                            + "sum2Speed, maxDistance, minDistance, maxTime, mintime, maxSpeed, minSpeed, numSamples)";
+                    String rows = "(" + recvTimeMs + ",'" + servicePath + "','" + entityId + "','" + entityType + "',"
+                            + location + ",0,0,0,0,0,0,0,0,0,-999999,999999,-999999,999999,-999999,999999,1)";
```

Comment:
What about using `Integer.MAX_VALUE` and `Integer.MIN_VALUE` here instead of `999999` and '-999999' respectively (or even `Float.MAX_VALUE` and `Float.MIN_VALUE` or `Double.MAX_VALUE` and `Double.MIN_VALUE`? :)